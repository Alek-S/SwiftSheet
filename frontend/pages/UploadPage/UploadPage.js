import React, { useState } from 'react';
import styled from 'styled-components';
import defaultStyle, { ErrorDialog } from '../../defaultStyle';
import Filedrop from '../../components/Filedrop/Filedrop';
import gql from 'graphql-tag';
import { Mutation } from 'react-apollo';
import { Redirect, Link } from 'react-router-dom';
import Papa from 'papaparse';

const UPLOAD_SHEET = gql`
	mutation UPLOAD_SHEET($sheetData: JSON!, $expireIn: Int!, $password: String) {
		createSheet(
			sheetData: $sheetData
			expireIn: $expireIn
			password: $password
		) {
			_id
		}
	}
`;

/**
 * @function
 * Page to upload file and set upload settings
 * @returns {JSX} <UploadPage />
 */
const UploadPage = () => {
	const [expireIn, setExpireIn] = useState(72);
	const [header, setHeader] = useState(true);
	const [disableSubmit, setDisableSubmit] = useState(true);
	const [redirect, setRedirect] = useState(false);
	const [file, setFile] = useState();
	const [password, setPassword] = useState('');
	const [wrongPassword, setWrongPassword] = useState(false);
	const [uploadErrorMessage, setErrorMessage] = useState('');

	const onCompleted = () => {
		setRedirect(true);
	};

	const renderRedirect = data => {
		const id = data && data.createSheet._id;
		if (redirect) {
			return <Redirect to={`/${id}`} />;
		}
	};

	const onDrop = (file, rejectedFile) => {
		if (
			file[0] &&
			file[0].name &&
			file[0].name.indexOf('.csv') + 4 === file[0].name.length
		) {
			setFile(file);
			setDisableSubmit(false);
		} else {
			console.error(
				`file rejected!: \n ${rejectedFile[0].name}\n ${rejectedFile[0].type}`
			);
		}
	};

	const checkPassword = pw => {
		const passing = (pw.length !== 0 && pw.length < 6) || pw.length > 70;
		setPassword(pw);
		setWrongPassword(passing);
	};

	const showErrorMessage = visible => {
		let message = '⚠️ Woops! Something went wrong.';

		if (header) {
			message += ' You sure first row is a header?';
		}

		setErrorMessage(message);
	};

	return (
		<StyledDiv>
			<Filedrop
				firstRowHeader={header}
				expireIn={expireIn}
				setHeader={setHeader}
				password={password}
				setPassword={checkPassword}
				setExpireIn={setExpireIn}
				onDrop={onDrop}
				wrongPassword={wrongPassword}
			/>
			<StyledForm disableSubmit={disableSubmit}>
				<Mutation mutation={UPLOAD_SHEET} onCompleted={onCompleted}>
					{(uploadSheet, { loading, error, data }) => {
						if (error) {
							showErrorMessage();
						}

						return (
							<div>
								{renderRedirect(data)}
								<input
									type="submit"
									onClick={e => {
										e.preventDefault();
										Papa.parse(file[0], {
											header: header,
											download: true,
											skipEmptyLines: false,
											complete: ({ data }) =>
												uploadSheet({
													variables: {
														sheetData: data,
														expireIn: parseInt(expireIn),
														password: password,
													},
												}),
										});
									}}
									value="Upload File"
									disabled={disableSubmit}
								/>
							</div>
						);
					}}
				</Mutation>
				<UploadError className={uploadErrorMessage ? 'true' : undefined}>
					{uploadErrorMessage}
				</UploadError>
			</StyledForm>
		</StyledDiv>
	);
};

const StyledDiv = styled(defaultStyle)`
	background-color: ${props => props.theme.color.background};
	padding-top: 7rem;
	height: calc(100vh - 100px);
`;

const StyledForm = styled.form`
	display: block;
	width: 60%;
	margin: auto;
	text-align: center;
	margin-top: 2rem;

	label {
		font-size: 1.1rem;
	}
	select {
		font-family: ${props => props.theme.font.main};
		font-size: 1rem;
		background-color: white;
		margin-left: 0.5rem;
		margin-right: 0.5rem;
		border: solid 1px ${props => props.theme.color.border};
	}

	input[type='submit'] {
		position: relative;
		cursor: ${props => (props.disableSubmit ? 'no-drop' : 'pointer')};
		color: ${props =>
			props.disableSubmit ? props.theme.color.border : 'white'};
		display: block;
		margin: auto;
		font-family: ${props => props.theme.font.main};
		font-weight: 400;
		font-size: 1.3rem;
		border: ${props =>
			props.disableSubmit ? `2px solid ${props.theme.color.border}` : 'none'};
		border-radius: 30px;
		padding: 10px 30px;
		background: ${props =>
			props.disableSubmit ? 'transparent' : props.theme.gradient.greenBlue};
		box-shadow: ${props => (props.disableSubmit ? '' : props.theme.boxShadow)};
		transition: all 0.5s;

		&:hover {
			background: ${props =>
				props.disableSubmit ? 'transparent' : props.theme.gradient.greenBlue};
			box-shadow: ${props =>
				props.disableSubmit ? '' : props.theme.boxShadowDark};
		}
	}
`;

const UploadError = styled(ErrorDialog)`
	min-width: 200px;
	width: fit-content;
	margin: auto;
	margin-top: 2rem;
`;

export default UploadPage;
