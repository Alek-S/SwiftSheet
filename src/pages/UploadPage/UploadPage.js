import React, { useState } from 'react';
import styled from 'styled-components';
import defaultStyle from '../../defaultStyle';
import Filedrop from '../../components/Filedrop/Filedrop';
import gql from 'graphql-tag';
import { Mutation } from 'react-apollo';

const UPLOAD_SHEET = gql`
	mutation UPLOAD_SHEET($sheetData: JSON!) {
		createSheet(sheetData: $sheetData) {
			_id
		}
	}
`;

/**
 * @function
 * Page to upload file and set upload settings
 * @returns {JSX} <UploadPage />
 */
const UploadPage = () => {
	const [expireIn, setExpireIn] = useState(72);
	const [header, setHeader] = useState(true);
	const [disableSubmit, setDisableSubmit] = useState(true);
	const [sheetData, setSheetData] = useState();

	/**
	 * @function
	 * Set expireIn state, in Hrs
	 * @param {object} e - dom event
	 */
	const _handleChange = e => {
		e.preventDefault();
		setExpireIn(e.target.value);
	};

	/**
	 * @function
	 * Toggle first row of CSV header boolean
	 * @param {object} e - dom event
	 */
	const _toggleHeader = e => {
		e.preventDefault();
		setHeader(!header);
	};

	/**
	 * @function
	 * Handle response from <FileDrop />
	 * @param {object} response - converted data from FileDrop
	 */
	const handleJSONData = response => {
		console.log(response.data);
		setSheetData(response.data);
		setDisableSubmit(false);
	};

	return (
		<StyledDiv>
			<Filedrop firstRowHeader={header} handleJSONData={handleJSONData} />
			<StyledForm disableSubmit={disableSubmit}>
				<Options>
					<label>
						Expires In:
						<select value={expireIn} onChange={_handleChange}>
							<option value={4}>4 Hours</option>
							<option value={8}>8 Hours</option>
							<option value={24}>1 Day</option>
							<option value={72}>3 Days</option>
							<option value={120}>5 Days</option>
						</select>
					</label>
					<HeaderToggle active={header}>
						<span>First Row Header:</span>
						<button onClick={_toggleHeader}>{header ? 'Yes' : 'No'}</button>
					</HeaderToggle>
				</Options>
				<Mutation mutation={UPLOAD_SHEET}>
					{(uploadSheet, { loading, error, data }) => (
						<input
							type="submit"
							onClick={e => {
								e.preventDefault();
								uploadSheet({
									variables: { sheetData },
								});
							}}
							value="Upload File"
							disabled={disableSubmit}
						/>
					)}
				</Mutation>
			</StyledForm>
		</StyledDiv>
	);
};

const StyledDiv = styled(defaultStyle)`
	background-color: ${props => props.theme.color.background};
	padding-top: 7rem;
	height: calc(100vh - 100px);
`;

const StyledForm = styled.form`
	display: block;
	width: 60%;
	margin: auto;
	text-align: center;
	margin-top: 2rem;

	label {
		font-size: 1.1rem;
	}
	select {
		font-family: ${props => props.theme.font.main};
		font-size: 1rem;
		background-color: white;
		margin-left: 0.5rem;
		margin-right: 0.5rem;
		border: solid 1px ${props => props.theme.color.border};
	}

	input[type='submit'] {
		position: relative;
		cursor: ${props => (props.disableSubmit ? 'no-drop' : 'pointer')};
		color: ${props =>
			props.disableSubmit ? props.theme.color.border : 'white'};
		display: block;
		margin: auto;
		margin-top: 4rem;
		font-family: ${props => props.theme.font.main};
		font-weight: 400;
		font-size: 1.3rem;
		border: ${props =>
			props.disableSubmit ? `2px solid ${props.theme.color.border}` : 'none'};
		border-radius: 30px;
		padding: 10px 30px;
		background: ${props =>
			props.disableSubmit ? 'transparent' : props.theme.gradient.greenBlue};
		box-shadow: ${props => (props.disableSubmit ? '' : props.theme.boxShadow)};
		transition: all 0.5s;

		&:hover {
			background: ${props =>
				props.disableSubmit ? 'transparent' : props.theme.gradient.greenBlue};
			box-shadow: ${props =>
				props.disableSubmit ? '' : props.theme.boxShadowDark};
		}
	}
`;

const HeaderToggle = styled.label`
	margin-left: 2rem;

	button {
		color: white;
		cursor: pointer;
		outline: none;
		background: ${props =>
			props.active
				? props.theme.gradient.greenBlue
				: props.theme.color.backgroundDarkest};
		box-shadow: ${props => (props.active ? props.theme.boxShadow : '')};
		border: none;
		font-family: ${props => props.theme.font.main};
		font-size: 1rem;
		border-radius: 4px;
		margin-left: 1rem;
		padding-top: 5px;
		padding-bottom: 5px;
		width: 70px;
	}
`;

const Options = styled.div`
	background-color: ${props => props.theme.color.backgroundDark};
	border-radius: 3px;
	padding-top: 1rem;
	padding-bottom: 1rem;
`;
export default UploadPage;
