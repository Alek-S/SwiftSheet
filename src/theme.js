/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
