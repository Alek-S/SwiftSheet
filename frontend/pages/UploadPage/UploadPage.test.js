import React from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
import { fireEvent, render, getByLabelText } from '@testing-library/react';
import wait from 'waait';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

import UploadPage from './UploadPage';
import theme from '../../theme';

describe('UploadPage', () => {
	test('matches the snapshot', () => {
		const wrapper = mount(
			<MockedProvider mocks={[]} addTypename={false}>
				<Router>
					<ThemeProvider theme={theme}>
						<UploadPage />
					</ThemeProvider>
				</Router>
			</MockedProvider>
		);

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test('it should start with upload button disabled', () => {
		const ui = render(
			<MockedProvider mocks={[]} addTypename={false}>
				<Router>
					<ThemeProvider theme={theme}>
						<UploadPage />
					</ThemeProvider>
				</Router>
			</MockedProvider>
		);

		const uploadButton = ui.getByLabelText('upload button');
		expect(uploadButton).toBeDisabled();
	});
});
