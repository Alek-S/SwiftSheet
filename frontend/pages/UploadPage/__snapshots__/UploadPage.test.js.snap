// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadPage matches the snapshot 1`] = `
.c1 {
  background-color: white;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  font-size: 0.9em;
}

.c8 {
  border-radius: 8px;
  box-shadow: 0px 5px 40px -5px rgba(0,0,0,0.15);
}

.c8 input {
  outline: none;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  background-color: #F1F3F4;
  border: 1px solid #F1F3F4;
  border-radius: 5rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c8 input::-webkit-input-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c8 input::-moz-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c8 input:-ms-input-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c8 input::placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c8 input:hover {
  border: solid 1px #d9d9d9;
}

.c8 input:focus,
.c8 input:active {
  border: solid 1px #46C8F5;
}

.c8 input.true {
  border: 1px solid #ED5050;
}

.c6 {
  text-align: center;
  opacity: 0;
  color: white;
  padding: 0.5rem 1rem;
  border-radius: 5px;
  background-color: #ED5050;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c6.true {
  opacity: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  background-color: #fff;
  border: dashed 2px #d9d9d9;
  border-radius: 5px;
  height: 250px;
  width: 70%;
  margin: auto;
  margin-top: 3rem;
}

.c3 img {
  display: block;
  height: 200px;
  width: 80%;
  margin: auto;
}

.c4 {
  color: #848484;
  font-size: 1.2rem;
  text-align: center;
  background-color: transparent;
}

.c7 {
  padding-top: 1.75rem;
  padding-bottom: 1.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 70%;
  margin: auto;
  text-align: center;
  margin-top: 3.5rem;
}

.c7 input {
  margin: 0 2rem 0 0.25rem;
  padding-left: 1rem;
  width: 140px;
  font-size: 0.9rem;
  font-weight: 400;
  height: 25px;
}

.c7 input::-webkit-input-placeholder {
  color: #BCBCBC;
}

.c7 input::-moz-placeholder {
  color: #BCBCBC;
}

.c7 input:-ms-input-placeholder {
  color: #BCBCBC;
}

.c7 input::placeholder {
  color: #BCBCBC;
}

.c7 select {
  margin-left: 0.25rem;
}

.c9 {
  margin-left: 2rem;
}

.c9 button {
  color: white;
  cursor: pointer;
  outline: none;
  background: linear-gradient(160deg,#57DACB 0%,#55A2ED 100%);
  box-shadow: 0px 5px 40px -5px rgba(0,0,0,0.3);
  border: none;
  font-family: 'Quicksand',sans-serif;
  font-size: 1rem;
  border-radius: 4px;
  margin-left: 1rem;
  padding-top: 5px;
  padding-bottom: 5px;
  width: 70px;
}

.c5 {
  min-width: 200px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin-top: 2rem;
}

.c0 {
  background-color: #fafafa;
  padding-top: 7rem;
  height: calc(100vh - 100px);
}

.c10 {
  display: block;
  width: 60%;
  margin: auto;
  text-align: center;
  margin-top: 2rem;
}

.c10 label {
  font-size: 1.1rem;
}

.c10 select {
  font-family: 'Quicksand',sans-serif;
  font-size: 1rem;
  background-color: white;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  border: solid 1px #d9d9d9;
}

.c10 input[type='submit'] {
  position: relative;
  cursor: no-drop;
  color: #d9d9d9;
  display: block;
  margin: auto;
  font-family: 'Quicksand',sans-serif;
  font-weight: 400;
  font-size: 1.3rem;
  border: 2px solid #d9d9d9;
  border-radius: 30px;
  padding: 10px 30px;
  background: transparent;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c10 input[type='submit']:hover {
  background: transparent;
}

.c11 {
  min-width: 200px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin: auto;
  margin-top: 2rem;
}

<MockedProvider
  addTypename={false}
  mocks={Array []}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "version": "2.4.12",
        "watchQuery": [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <ThemeProvider
          theme={
            Object {
              "boxShadow": "0px 5px 40px -5px rgba(0,0,0,0.3)",
              "boxShadowDark": "0px 5px 40px -2px rgba(0,0,0,0.5)",
              "boxShadowLight": "0px 5px 40px -5px rgba(0,0,0,0.15)",
              "color": Object {
                "aqua": "#68F6BA",
                "background": "#fafafa",
                "backgroundDark": "#EAEAEA",
                "backgroundDarkest": "#BCBCBC",
                "blue": "#46C8F5",
                "border": "#d9d9d9",
                "darkBlue": "#2E8AA8",
                "green": "#D8E13F",
                "input": "#F1F3F4",
                "lightBlue": "#97E4FF",
                "lightRed": "#FF8D8D",
                "lightText": "#848484",
                "orange": "#ffbf80",
                "red": "#ED5050",
                "text": "#404040",
                "vividGreen": "#E3FC52",
              },
              "font": Object {
                "header": "'Poiret One', cursive",
                "main": "'Quicksand', sans-serif",
              },
              "gradient": Object {
                "green": "linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)",
                "greenBlue": " linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)",
                "light": "linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)",
                "lightRed": "linear-gradient(to bottom, #f39191 0%, #ef6262 100%)",
                "red": "linear-gradient(to bottom, #f07878 0%, #ED5050 100%)",
              },
            }
          }
        >
          <UploadPage>
            <UploadPage__StyledDiv>
              <defaultStyle
                className="c0"
              >
                <div
                  className="c0 c1"
                >
                  <Filedrop
                    expireIn={72}
                    firstRowHeader={true}
                    onDrop={[Function]}
                    password=""
                    setExpireIn={[Function]}
                    setHeader={[Function]}
                    setPassword={[Function]}
                    wrongPassword={false}
                  >
                    <Filedrop__StyledSection>
                      <section
                        className="c2"
                      >
                        <Filedrop__StyledDropzone
                          accept="text/csv"
                          maxSize={2500000}
                          minSize={8}
                          multiple={false}
                          onDragEnter={[Function]}
                          onDragLeave={[Function]}
                          onDrop={[Function]}
                          onDropRejected={[Function]}
                        >
                          <t
                            accept="text/csv"
                            className="c3"
                            disableClick={false}
                            disablePreview={false}
                            disabled={false}
                            getDataTransferItems={[Function]}
                            inputProps={Object {}}
                            maxSize={2500000}
                            minSize={8}
                            multiple={false}
                            onDragEnter={[Function]}
                            onDragLeave={[Function]}
                            onDrop={[Function]}
                            onDropRejected={[Function]}
                            preventDropOnDocument={true}
                          >
                            <div
                              aria-disabled={false}
                              className="c3"
                              onClick={[Function]}
                              onDragEnter={[Function]}
                              onDragLeave={[Function]}
                              onDragOver={[Function]}
                              onDragStart={[Function]}
                              onDrop={[Function]}
                              style={
                                Object {
                                  "position": "relative",
                                }
                              }
                            >
                              <img
                                alt="upload-cloud-logo"
                                src="./assets/images/upload-cloud-light.svg"
                              />
                              <Filedrop__StyledText
                                dragging={false}
                              >
                                <defaultStyle
                                  className="c4"
                                  dragging={false}
                                >
                                  <div
                                    className="c4 c1"
                                  >
                                    Drag
                                     CSV file here
                                  </div>
                                </defaultStyle>
                              </Filedrop__StyledText>
                              <input
                                accept="text/csv"
                                autoComplete="off"
                                disabled={false}
                                multiple={false}
                                onChange={[Function]}
                                style={
                                  Object {
                                    "bottom": 0,
                                    "left": 0,
                                    "opacity": 0.00001,
                                    "pointerEvents": "none",
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                  }
                                }
                                type="file"
                              />
                            </div>
                          </t>
                        </Filedrop__StyledDropzone>
                        <Filedrop__UploadError>
                          <defaultStyle__ErrorDialog
                            className="c5"
                          >
                            <div
                              className="c5 c6"
                            />
                          </defaultStyle__ErrorDialog>
                        </Filedrop__UploadError>
                        <Filedrop__Options
                          aria-label="preferences"
                          role="form"
                        >
                          <defaultStyle__Card
                            aria-label="preferences"
                            className="c7"
                            role="form"
                          >
                            <defaultStyle
                              aria-label="preferences"
                              className="c7 c8"
                              role="form"
                            >
                              <div
                                aria-label="preferences"
                                className="c7 c8 c1"
                                role="form"
                              >
                                <section
                                  aria-label="set password"
                                >
                                  Password:
                                  <input
                                    autoComplete="new-password"
                                    onChange={[Function]}
                                    placeholder="enter password"
                                    type="password"
                                    value=""
                                  />
                                </section>
                                <section
                                  aria-label="set expiration"
                                >
                                  Expires In:
                                  <select
                                    onChange={[Function]}
                                    value={72}
                                  >
                                    <option
                                      value={1}
                                    >
                                      1 Hour
                                    </option>
                                    <option
                                      value={4}
                                    >
                                      4 Hours
                                    </option>
                                    <option
                                      value={8}
                                    >
                                      8 Hours
                                    </option>
                                    <option
                                      value={24}
                                    >
                                      1 Day
                                    </option>
                                    <option
                                      value={72}
                                    >
                                      3 Days
                                    </option>
                                    <option
                                      value={120}
                                    >
                                      5 Days
                                    </option>
                                  </select>
                                </section>
                                <section
                                  aria-label="toggle first row header"
                                >
                                  <Filedrop__HeaderToggle
                                    active={true}
                                  >
                                    <label
                                      className="c9"
                                    >
                                      <span>
                                        First Row Header:
                                      </span>
                                      <button
                                        onClick={[Function]}
                                      >
                                        Yes
                                      </button>
                                    </label>
                                  </Filedrop__HeaderToggle>
                                </section>
                              </div>
                            </defaultStyle>
                          </defaultStyle__Card>
                        </Filedrop__Options>
                        <Filedrop__UploadError>
                          <defaultStyle__ErrorDialog
                            className="c5"
                          >
                            <div
                              className="c5 c6"
                            >
                              ⚠️ Check Password Length
                            </div>
                          </defaultStyle__ErrorDialog>
                        </Filedrop__UploadError>
                      </section>
                    </Filedrop__StyledSection>
                  </Filedrop>
                  <UploadPage__StyledForm
                    disableSubmit={true}
                  >
                    <form
                      className="c10"
                    >
                      <Mutation
                        mutation={
                          Object {
                            "definitions": Array [
                              Object {
                                "directives": Array [],
                                "kind": "OperationDefinition",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "UPLOAD_SHEET",
                                },
                                "operation": "mutation",
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [
                                        Object {
                                          "kind": "Argument",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "sheetData",
                                          },
                                          "value": Object {
                                            "kind": "Variable",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "sheetData",
                                            },
                                          },
                                        },
                                        Object {
                                          "kind": "Argument",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "expireIn",
                                          },
                                          "value": Object {
                                            "kind": "Variable",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "expireIn",
                                            },
                                          },
                                        },
                                        Object {
                                          "kind": "Argument",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "password",
                                          },
                                          "value": Object {
                                            "kind": "Variable",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "password",
                                            },
                                          },
                                        },
                                      ],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "createSheet",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "_id",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                                "variableDefinitions": Array [
                                  Object {
                                    "defaultValue": undefined,
                                    "directives": Array [],
                                    "kind": "VariableDefinition",
                                    "type": Object {
                                      "kind": "NonNullType",
                                      "type": Object {
                                        "kind": "NamedType",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "JSON",
                                        },
                                      },
                                    },
                                    "variable": Object {
                                      "kind": "Variable",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "sheetData",
                                      },
                                    },
                                  },
                                  Object {
                                    "defaultValue": undefined,
                                    "directives": Array [],
                                    "kind": "VariableDefinition",
                                    "type": Object {
                                      "kind": "NonNullType",
                                      "type": Object {
                                        "kind": "NamedType",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "Int",
                                        },
                                      },
                                    },
                                    "variable": Object {
                                      "kind": "Variable",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "expireIn",
                                      },
                                    },
                                  },
                                  Object {
                                    "defaultValue": undefined,
                                    "directives": Array [],
                                    "kind": "VariableDefinition",
                                    "type": Object {
                                      "kind": "NamedType",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "String",
                                      },
                                    },
                                    "variable": Object {
                                      "kind": "Variable",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "password",
                                      },
                                    },
                                  },
                                ],
                              },
                            ],
                            "kind": "Document",
                            "loc": Object {
                              "end": 187,
                              "start": 0,
                            },
                          }
                        }
                        onCompleted={[Function]}
                      >
                        <div>
                          <input
                            disabled={true}
                            onClick={[Function]}
                            type="submit"
                            value="Upload File"
                          />
                        </div>
                      </Mutation>
                      <UploadPage__UploadError>
                        <defaultStyle__ErrorDialog
                          className="c11"
                        >
                          <div
                            className="c11 c6"
                          />
                        </defaultStyle__ErrorDialog>
                      </UploadPage__UploadError>
                    </form>
                  </UploadPage__StyledForm>
                </div>
              </defaultStyle>
            </UploadPage__StyledDiv>
          </UploadPage>
        </ThemeProvider>
      </Router>
    </BrowserRouter>
  </ApolloProvider>
</MockedProvider>
`;
